### DISCUSSION TITLE
Grafana and Homer

### DISCUSSION THREAD

**Discussion by chabeaumont** [sipcapture/homer#525] [Q&A] [answered]
Hi everyone,

I'm trying to get some statistics with Homer and Grafana with this template : https://grafana.com/orgs/lmangani
but didn't find a lot of informations how to make it. 

I tried adding Prometheus as datasource in Grafana but always get this answer : 

![image](https://user-images.githubusercontent.com/98812038/152954720-c00d403e-31ee-4eb6-875a-18a274eca2d2.png)

![image](https://user-images.githubusercontent.com/98812038/152954807-9ab6a385-3884-49ed-876a-a8e34510d35b.png)

I've no issue with the : "wget http://10.0.0.70:8086/metrics" on the grafana server side.

am I forgetting something ?

**Comment by lmangani** [comment, marked as answer]
Hello @chabeaumont 

The configuration seems incorrect. The /metrics endpoint to be scraped by Prometheus is served by heplify-server using the PROMADDR settings, ie:
```
      - "HEPLIFYSERVER_PROMADDR=0.0.0.0:9096"
```

So the correct scraper config would look like this:
```
  - job_name: 'heplify-server'
    scrape_interval: 5s
    static_configs:
      - targets: ['heplify-server:9096']
```

I'm not sure what 10.0.0.70 might be in your setup, but the port doesn't make sense. 
A full example is available [here](https://github.com/sipcapture/homer7-docker/tree/master/heplify-server/hom7-prom-all). Please confirm if this helps.

**Reply by chabeaumont** [reply]
Hello Lorenzo,

Thanks for your answer. 
10.0.0.70 is my homer/heplify server. 
I tried this but failed with the same error : http://10.0.0.70:9096

"wget http://10.0.0.70:9096/metrics" working fine on my grafana server.
I also tried to install grafana on local on my 10.0.0.70 server but still have the same error with http://127.0.0.1:9096 or http://localhost:9096.

**Comment by lmangani** [comment]
@chabeaumont sounds like a local network issue to me, is there a firewall or proxy in front of this deployment?
Please provide the exact errors and response codes with as much detail as possible in order to get any applicable suggestion.

**Reply by chabeaumont** [reply]
No it's local network. No proxy, no firewall. I've the same issue with grafana installed on local on my heplify server. 

First configuration : 
local network : no proxy no firewall.

Debian 11 : Grafana [v8.3.3] : 10.0.0.61
Debian 10 : Homer/Heplify [7.10.16] : 10.0.0.70

From grafana server : 

```
root@DIGPRTESUP02:~# wget -O- http://10.0.0.70:9096/metrics
--2022-02-08 11:04:40--  http://10.0.0.70:9096/metrics
Connexion à 10.0.0.70:9096… connecté.
requête HTTP transmise, en attente de la réponse… 200 OK

go_gc_duration_seconds A summary of the pause duration of garbage collection cycles.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 9.513e-06
go_gc_duration_seconds{quantile="0.25"} 1.1224e-05
[...]
```

![image](https://user-images.githubusercontent.com/98812038/152965401-435f5455-1cf2-4da2-999e-24fe654ce266.png)

![image](https://user-images.githubusercontent.com/98812038/152965492-7e87dcc9-bd32-4fc8-be68-0c8ab21de6d3.png)


Second configuration : 

Debian 10 : Homer/Heplify [7.10.16]  and **Fresh install Grafana [v7.4.3]**: 10.0.0.70

![image](https://user-images.githubusercontent.com/98812038/152966104-c7c1c7b5-db4f-4907-a8d9-d7a1516ef58e.png)

**Reply by chabeaumont** [reply]
```
root@DIGPRHOMER01:~# netstat -tulpn | grep LISTEN
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      591/sshd
tcp        0      0 127.0.0.1:8088          0.0.0.0:*               LISTEN      13056/influxd
tcp        0      0 127.0.0.1:5432          0.0.0.0:*               LISTEN      623/postgres
tcp6       0      0 :::8086                 :::*                    LISTEN      13056/influxd
tcp6       0      0 :::22                   :::*                    LISTEN      591/sshd
tcp6       0      0 :::3000                 :::*                    LISTEN      13489/grafana-serve
tcp6       0      0 :::8888                 :::*                    LISTEN      469/chronograf
tcp6       0      0 :::9080                 :::*                    LISTEN      658/homer-app
tcp6       0      0 ::1:5432                :::*                    LISTEN      623/postgres
tcp6       0      0 :::9060                 :::*                    LISTEN      893/heplify-server
tcp6       0      0 :::9092                 :::*                    LISTEN      551/kapacitord
tcp6       0      0 :::9096                 :::*                    LISTEN      893/heplify-server
```

**Comment by lmangani** [comment]
Wait a second @chabeaumont Grafana is NOT supposed to connect to any of our elements, Prometheus is.

> heplify-server <--- Prometheus <---- Grafana

So your Prometheus scraper needs to connect to heplify-server, and Grafana in turn connects to the Prometheus API

I hope this clears the doubts up. Follow the docker examples as they contain all the relevant configuration at once.

**Reply by chabeaumont** [reply]
Ok, just to be sure I understand well :
I need to connect Graphana to Promotheus Server that is not installed by default with Homer ? So I need first to install the Prometheus server, configure it to be feed by Heplify and connect Grafana to Prometheus Server ?
Is there any documentation about this part? If not I will be happy to help by writing it if I success.

**Comment by adubovikov** [comment]
@chabeaumont heplify-server has only API to provide internal statistics. To display the data you need Prometheus Server or InfluxDB and pull/push data from heplify-server to your Prometheus/InfluxDB 

As I see you have already InfluxDB - please add the influxdb source and get date from there

**Reply by chabeaumont** [reply]
Thank you for your answer, I already add InfluxDB in Grafana, but I'm pretty novice in Grafana and want to use the template from [here](https://grafana.com/orgs/lmangani) to start with :).
